### DISCUSSION TITLE
Ability to customize controls?

### DISCUSSION THREAD

**Discussion by nooblyf** [fabricjs/fabric.js#9174] [Q&A] [answered]
Is it possible to completely customize controls of everything. I want to make it look something like this

![ss](https://github.com/fabricjs/fabric.js/assets/54323745/92b8f56c-fa4a-463d-9b9a-2d8c3df39bfa)

**Comment by lh-sn** [comment, marked as answer]
Yeah, you can completely customize the controls.

You can play here: https://codesandbox.io/p/sandbox/fabric-custom-controls-demo-794x8l

**Reply by nooblyf** [reply]
Cool, thank you!!

**Reply by niketaOptimumnbrew** [reply]
> Yeah, you can completely customize the controls.
> 
> You can play here: https://codesandbox.io/p/sandbox/fabric-custom-controls-demo-794x8l

Hello @ShaMan123 

Could you please check why cursor not show when hover on middle-left , middle-right , middle-top and middle-bottom's top and bottom section?

Please check below image:
![image](https://github.com/user-attachments/assets/27fd9b42-cc3c-4fe3-b8f2-b277e688ee9f)

Cursor show when I hover on center section of control but not show when hover on top and bottom section.

Can you please help me what's wrong in this?

**Reply by ShaMan123** [reply]
I suggest you render the oCoords[key].corner values to see if you have correct calculation

**Reply by niketaOptimumnbrew** [reply]
> I suggest you render the oCoords[key].corner values to see if you have correct calculation

@ShaMan123 Thanks for quick reply,
but I'm not getting you from where can I render oCoords[key].corner values ?

**Reply by ShaMan123** [reply]
Object.values(myObj.oCoords).forEach(coord => renderThis(coord.corner))

**Reply by niketaOptimumnbrew** [reply]
> Object.values(myObj.oCoords).forEach(coord => renderThis(coord.corner))

@ShaMan123 Could you please help me little bit more?

I don't know where to put above code?

**Reply by ShaMan123** [reply]
Wherever
Open devtools or in the after render hook

**Reply by niketaOptimumnbrew** [reply]
> Object.values(myObj.oCoords).forEach(coord => renderThis(coord.corner))

@ShaMan123 above solution not working.
Could you please again look into it?

**Reply by ShaMan123** [reply]
It is pseudo code

**Reply by asturur** [reply]
@niketaOptimumnbrew that control is probably drawn larger than what is its interaction area.
You have to check either the specific control size properties that you can see in the control class or play with the object cornerSize property.

**Reply by xo-o** [reply]
Share a reproduction...
